src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,484 @@
+        self._invalidate_delta_table_cache(feature_key)
+
+        # Strategy: Use Polars write_delta for eager DataFrames (native support),
+        # convert to PyArrow for lazy frames and use deltalake API (more flexible)
+        if isinstance(df, pl.LazyFrame):
+            # Check if streaming mode is enabled
+            if self.streaming_chunk_size is not None:
+                # Streaming mode: write batches incrementally without full collection
+                # This enables processing datasets larger than RAM at the cost of:
+                # - Slower performance (sink_batches overhead)
+                # - More Delta versions (one per batch)
+                def write_batch(batch_df: pl.DataFrame) -> bool:
+                    arrow_table = batch_df.to_arrow()
+                    self._deltalake.write_deltalake(
+                        table_uri,
+                        arrow_table,
+                        mode="append",
+                        **self._delta_write_options,
+                        storage_options=self.storage_options or None,
+                    )
+                    return False  # Continue processing all batches
+
+                df.sink_batches(write_batch, chunk_size=self.streaming_chunk_size)  # pyright: ignore[reportCallIssue]
+            else:
+                # Default mode: collect then write (faster for normal-sized datasets)
+                arrow_table = df.collect().to_arrow()
+                self._deltalake.write_deltalake(
+                    table_uri,
+                    arrow_table,
+                    mode="append",
+                    **self._delta_write_options,
+                    storage_options=self.storage_options or None,
+                )
+        else:
+            # For eager DataFrames: use Polars native write_delta
+            # This is efficient and leverages Polars' built-in Delta support
+            df.write_delta(
+                table_uri,
+                mode="append",
+                delta_write_options=self._delta_write_options,
+                storage_options=self.storage_options or None,
+            )
+
+    def _drop_feature_metadata_impl(self, feature_key: FeatureKey) -> None:
+        """Drop Delta table for the specified feature using soft delete.
+
+        Uses Delta's delete operation which marks rows as deleted in the transaction log
+        rather than physically removing files.
+        """
+        # Check if table exists first
+        if not self._table_exists(
+            self._feature_uri(feature_key), feature_key=feature_key
+        ):
+            return
+
+        # Load the Delta table
+        delta_table = self._get_delta_table(feature_key, include_files=False)
+
+        # Use Delta's delete operation - soft delete all rows
+        # This marks rows as deleted in transaction log without physically removing files
+        delta_table.delete()
+        self._invalidate_delta_table_cache(feature_key)
```

Comment:
Why should we be invalidating this cache here? The DeltaTable object is still valid.